### PR TITLE
Add domainname test

### PR DESCRIPTION
**Pull request by higuruchi**
Add domainname test.
nix does't have getdomainname function, So used libc crate.

Signed-off-by: higuruchi <fumiya2324@gmail.com>

### CODE REVIEW

**Comment by utam0k** [comment]
@higuruchi Thanks for your cool PR, but we have to wait for runc side. https://github.com/opencontainers/runc/pull/3600

**Comment by yihuaf** [comment]
> @higuruchi Thanks for your cool PR, but we have to wait for runc side. [opencontainers/runc#3600](https://github.com/opencontainers/runc/pull/3600)

@utam0k I see the runc PR is now merged. Should we get this PR merged as well? @higuruchi do you have time to follow up? If not, I can help to clean this PR up and get it merged.

**Comment by utam0k** [comment]
@yihuaf We neet to wait for the release.  Thanks for your catch-up 🙏

**Comment by YJDoc2** [comment]
@utam0k I think we can proceed with now? Or is runc release with this is still not done?

**Comment by utam0k** [comment]
> @utam0k I think we can proceed with now? Or is runc release with this is still not done?

I think we can process this PR!

**Comment by YJDoc2** [comment]
@higuruchi let me know if you can continue working on this for re-basing and fixing conflicts. In case you are busy with something else or cannot work on this, I can take it up so let me know :+1:

**Comment by higuruchi** [comment]
I can continue this work. Thank you for letting me know.

File: tests/rust-integration-tests/runtimetest/src/main.rs [hunk 1]
```diff
@@ -30,6 +30,11 @@ fn main() {
         "readonly_paths" => tests::validate_readonly_paths(&spec),
         "set_host_name" => tests::validate_hostname(&spec),
         "mounts_recursive" => tests::validate_mounts_recursive(&spec),
+        "domainname_test" => tests::validate_domainname(&spec),
+        _ => eprintln!(
+            "error due to unexpected execute test name: {}",
+            execute_test
+        ),
```
Comment (YJDoc2) [inline tests/rust-integration-tests/runtimetest/src/main.rs:37]: I think this got duplicated in the master merge, and clippy is giving error...

**Comment by higuruchi** [comment]
@YJDoc2 

integration_tests_validation fails because the latest version of runc, v1.1.8, seems does not include [domain name pull request](https://github.com/opencontainers/runc/pull/3600). If i run the tests on the main branch binary, they will all pass. Wait for the  [domain name pull request](https://github.com/opencontainers/runc/pull/3600) to be added to the new release?

**Comment by YJDoc2** [comment]
I think they are waiting for a minor release to roll the domainname feature out, and recent releases have been patch , so they didn't include it. I guess we will simply have to wait. @utam0k wdyt? Should we instead move to the main branch with a fixed sha or only use releases? I feel releases would be better because that is what actually being shipped.

**Comment by higuruchi** [comment]
> @higuruchi May I ask you to rebase it to current master and do the following to fix the failing runc test : keep everything same, but comment out the addition of the test in main.rs . I'll make it work for both in a separate PR later. If you are busy with something else, I can take this up and do the necessary changes, let me know. Thanks a lot 🙏

@YJDoc2 
I can't make any changes to youki at the moment, so would you mind if I ask you to make a correction?

**Comment by YJDoc2** [comment]
>  I can't make any changes to youki at the moment, so would you mind if I ask you to make a correction?

No worries, I'll take care of this. Thanks a lot for the contributions :pray:

**Comment by higuruchi** [comment]
> > I can't make any changes to youki at the moment, so would you mind if I ask you to make a correction?
> 
> No worries, I'll take care of this. Thanks a lot for the contributions 🙏

Thanks a lot 🙏

**Review by utam0k — APPROVED** [review]
LGTM. Please fix the conflict.

**Comment by YJDoc2** [comment]
Thanks a lot @utam0k for review!
Thanks @higuruchi for adding this, and apologies, it took us so long to get this merged :pray:

**Pull request merged**